homeassistant/bootstrap.py

```diff
@@ -673,6 +674,21 @@ async def _async_set_up_integrations(
         requirements.async_load_installed_versions(hass, needed_requirements),
         "check installed requirements",
     )
+    return domains_to_setup, integration_cache
+
+
+async def _async_set_up_integrations(
+    hass: core.HomeAssistant, config: dict[str, Any]
+) -> None:
+    """Set up all the integrations."""
+    hass.data[DATA_SETUP_STARTED] = {}
+    setup_time: dict[str, timedelta] = hass.data.setdefault(DATA_SETUP_TIME, {})
+
+    watch_task = asyncio.create_task(_async_watch_pending_setups(hass))
+
+    domains_to_setup, integration_cache = await _async_resolve_domains_to_setup(
+        hass, config
+    )
```

Comment:
Why do you split thus function up in two?

Reply:
It was getting too complex to follow so I broke it down